src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -592,6 +598,40 @@ void GoogleCalendarSyncAdaptor::finalCleanup()
     if (succeededAccounts.size()) {
         setLastSyncSuccessful(succeededAccounts);
     }
+
+    // Delete any events which are not associated with a notebook.
+    // These events are ghost events, caused by a bug which previously
+    // existed in the sync adapter code due to mkcal deleteNotebook semantics.
```

Comment:
Added code to ensure we only do the orphan cleanup check once (uses gcal.ini key to determine whether to perform the check or not, so can be triggered by removing that file (or setting the key there to false).